flink-python/src/main/java/org/apache/flink/python/AbstractPythonFunctionRunner.java

```diff
@@ -176,10 +179,29 @@ public void open() throws Exception {
 		Struct pipelineOptions = PipelineOptionsTranslation.toProto(portableOptions);
 
 		jobBundleFactory = createJobBundleFactory(pipelineOptions);
-		stageBundleFactory = jobBundleFactory.forStage(createExecutableStage());
+		stageBundleFactory = createStageBundleFactory();
 		progressHandler = BundleProgressHandler.ignored();
 	}
 
+	/**
+	 * To make the error messages more user friendly, read the boot logs from the temp log file and
+	 * throw an exception directly.
+	 */
+	private StageBundleFactory createStageBundleFactory() throws Exception {
+		try {
+			return jobBundleFactory.forStage(createExecutableStage());
+		} catch (Exception e) {
+			byte[] output =
+				Files.readAllBytes(
+					new File(((ProcessPythonEnvironmentManager) environmentManager).getBaseDirectory()
+						+ "/flink-python-udf-boot.log").toPath());
```

Comment:
What about changing `/flink-python-udf-boot.log` to `File.separator + "flink-python-udf-boot.log`?